### ISSUE TITLE
"Reveal in file tree" doesn't actually reveal file

### ISSUE THREAD

**Issue by ffxsam** [facebook/nuclide#697]
### Issue and Steps to Reproduce

Right clicking a tab and choosing "Reveal in File Tree" or using cmd-shift-| shortcut has no effect.
### Expected Behavior

File tree should auto-expand folders as needed and highlight corresponding file.
### Actual Behavior

Nothing.
### Versions
- Atom: 1.10.2
- Nuclide: 0.166
- Client OS: Mac OS 10.11.6
- Server OS (optional):
### Additional Details
- Installed packages (`apm ls --installed`):

```
[36mCommunity Packages (21)[39m /Users/samh/.atom/packages
├── autocomplete-modules@1.6.1
├── expose@0.12.1
├── git-time-machine@1.5.3
├── goto-definition@1.1.19
├── haskell-grammar@0.4.0
├── highlight-selected@0.11.2
├── js-refactor@0.7.3
├── language-babel@2.37.0
├── language-ini@1.16.0
├── language-ocaml@1.1.2
├── language-swift@0.4.0
├── language-thrift@1.0.2
├── linter@1.11.16
├── linter-eslint@7.3.0
├── merge-conflicts@1.4.4
├── minimap@4.25.0
├── nuclide@0.166.0
├── refactor@0.9.2
├── sort-lines@0.14.0
├── todo-show@1.7.0
└── tool-bar@1.0.1

└── (empty)
```

**Comment by shushz** [comment]
Does your project path contain symlinks? If yes, could you try mounting on a path without?
There's a KP that this functionality doesn't work on project paths with symlinks in them (maybe added via `atom -a <path>`.)

**Comment by ffxsam** [comment]
It works in a new folder structure I set up that has no symlinks. So it appears that's the issue. But my project has symlinks I can't just get rid of (in `node_modules` and `.meteor`).

**Comment by shushz** [comment]
Reveal still works for everything that doesn't have a symlink in its path. So if you can just mount the project root by its resolved path most of the files, I guess, should be revealable.
Yes, it still won't work for the files under the symlinks in `node_modules` and `.meteor`, but it's a start.
Unfortunately, at the moment at least, Atom does not provide a way to resolve the symlink path for a TextEditor, but hopefully it'll be added (see https://github.com/atom/atom/issues/9879). Once it's in we'll be able to use that API to implement the reveal functionality properly for files under symlinks as well.

**Comment by ffxsam** [comment]
>Reveal still works for everything that doesn't have a symlink in its path

I'm afraid not. My open file is something in `imports/components/...etc` and there are no symlinks anywhere in there, yet the feature fails to work.

**Comment by shushz** [comment]
Could you try the following when such a file is opened in the editor?
* Open the dev tools console (Cmd-Opt-i)
* Type `atom.workspace.getActiveTextEditor().getPath()`
* Then, in the file-tree, locate the file that should have been revealed and using the dev tools' console inspect its element (Cmd-Shift-C and click on the file in the file-tree)
* Look for the `data-path` property  in the element. Is it identical to what the `...getPath()` command printed in the console?

From what I've seen the failure to "reveal" is related to these two being different (somewhat simplified. Just editing the `data-path` wouldn't make it work. It's the internal data structures that use the same key that ends up being in the `data-path`)

**Comment by ffxsam** [comment]
Aha, that's it! It's because all my web dev work is in `~/Sites` which is actually a symlink to `/Volumes/SuperData/Sites`.

**Comment by maxbrunsfeld** [comment]
I believe this Atom bug will be fixed in 1.14.

**Comment by shushz** [comment]
Thanks, @maxbrunsfeld. OK. I'm closing this one.
